### ISSUE TITLE
failing to parse due to invalid symbol error

### ISSUE THREAD

**Issue by lread** [borkdude/clj-kondo#378]
**version**
clj-kondo v2019.07.24-alpha

**platform**
macOS v10.14.6

**problem**
clj-kondo is failing with invalid symbol error

**repro**
Given the following `src/borkdude.clj` source:
```clojure
(ns borkdude)

(defn -main []
  (println 'org/clojure/math.numeric-tower))
```
Sanity check:
`clj -m borkdude` outputs:
```
org/clojure/math.numeric-tower
```

**expected behavior**
`clj-kondo --lint src` outputs no errors.

**actual behavior**
`clj-kondo --lint src` outputs:
```
src/borkdude.clj:0:0: error: can't parse src/borkdude.clj, Invalid symbol: org/clojure/math.numeric-tower.
linting took 3ms, errors: 1, warnings: 0
```

**Comment by yuhan0** [comment]
`'org/clojure/math.numeric-tower` is actually an illegal symbol according to the language spec:
https://clojure.org/reference/reader#_symbols

I think the current Clojure reader just happens not to enforce that rule, whereas clj-kondo's parser is stricter. And it may change in the future, see https://clojure.atlassian.net/browse/CLJ-1530

The expected behaviour should be to output a lint error pointing to the actual location of the symbol, i.e.
```
src/borkdude.clj:4:11: error: Invalid symbol: org/clojure/math.numeric-tower.
```

Here are some other read-time errors that cause the linter to fail for the entire file and output a "can't parse" error at 0:0 

```clojure
clojure.core/ ; <= invalid symbol
:             ; <= invalid keyword
10z           ; <= invalid number
```


I might try tackling this in the next day or two :)

**Comment by borkdude** [comment]
@yuhan0 Actually I made a similar patch for keywords to let clj-kondo be less strict about it, since Clojure also accepts it right now and someone was using this to represent CSS using the garden library.

See: https://github.com/borkdude/clj-kondo/blob/b9d52ff9e0109bb55e4c490beb767ddebbf25870/parser/clj_kondo/impl/rewrite_clj/parser/keyword.clj#L1

I think the same behavior is desirable for symbols to be consistent. Once Clojure itself enforces it, we can also enforce it in this linter.

Regardless, the location of the error can be improved.

**Comment by disalvjn** [comment]
I'm getting `Can't parse <stdin>, Key must be integer` on the following file:

```clj
(ns name)

(def x (String. "hello"))
```

Removing the `String.` resolves the problem -- it looks like it's the `.` at the end of the symbol causing the problem. This strikes me as more serious than the other examples in this thread, because I think it's valid per the language spec.

I'm using the v2020.7.29 VSCode extension.

**Comment by borkdude** [comment]
@disalvjn Please try to lint this file without any of your project config. Your project config is likely the problem.

**Comment by disalvjn** [comment]
@borkdude  You're right; the issue was with my project config. I had:

```clj
{:linters 
  {:unresolved-symbol [(faconne.core/transform) (cats.core/mlet)]}}
```

when I needed to wrap the vector in :exclude like
```clj
{:linters 
 {:unresolved-symbol 
    {:exclude [(faconne.core/transform) (cats.core/mlet)]}}}
```

Thanks for your speedy response and your help!

**Comment by borkdude** [comment]
@lread Can we close this issue or do you think clj-kondo should handle this differently?

**Comment by lread** [comment]
Yeah, I think we can close, thanks for asking.

FWIW, I've decided not to support this illegal syntax, at least for now, in rewrite-cljc.

**Comment by sogaiu** [comment]
FYI: https://github.com/carocad/parcera/pull/94#issuecomment-706546263
